From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aq17 <aqiu@pulumi.com>
Date: Thu, 25 May 2023 10:33:38 -0700
Subject: [PATCH] Set user agent in client options

Use Pulumi-specific user agent which includes the current provider version number.

diff --git a/internal/common/client_options.go b/internal/common/client_options.go
index 99352e4d62..5006e975b0 100644
--- a/internal/common/client_options.go
+++ b/internal/common/client_options.go
@@ -101,6 +101,8 @@ func userAgent(userAgent, tfVersion, partnerID string, disableTerraformPartnerID
 	if features.FivePointOhBeta() {
 		providerUserAgent = fmt.Sprintf("%s terraform-provider-azurerm/%s+5.0-beta", tfUserAgent, version.ProviderVersion)
 	}
+	// FORK: this gives us the ability to add a Pulumi Specific user agent
+	providerUserAgent = fmt.Sprintf("pulumi-azure/%s", version.ProviderVersion)
 	userAgent = strings.TrimSpace(fmt.Sprintf("%s %s", userAgent, providerUserAgent))
 
 	// append the CloudShell version to the user agent if it exists
@@ -109,11 +111,11 @@ func userAgent(userAgent, tfVersion, partnerID string, disableTerraformPartnerID
 	}
 
 	// only one pid can be interpreted currently
-	// hence, send partner ID if present, otherwise send Terraform GUID
+	// hence, send partner ID if present, otherwise send Pulumi GUID
 	// unless users have opted out
 	if partnerID == "" && !disableTerraformPartnerID {
-		// Microsoft’s Terraform Partner ID is this specific GUID
-		partnerID = "222c6c49-1b0a-5959-a213-6608f9eb8820"
+		// FORK: Microsoft’s Pulumi Partner ID is this specific GUID
+		partnerID = "a90539d8-a7a6-5826-95c4-1fbef22d4b22"
 	}
 
 	if partnerID != "" {
